data_types/softlayer_virtual_guest.go

```diff
@@ -54,31 +54,29 @@ type SoftLayer_Virtual_Guest_Template_Parameters struct {
 	Parameters []SoftLayer_Virtual_Guest_Template `json:"parameters"`
 }
 
+
+//According the SL API Doc(http://sldn.softlayer.com/reference/services/SoftLayer_Virtual_Guest/createObject),
+//added some miss properties in SoftLayer_Virtual_Guest_Template.
 type SoftLayer_Virtual_Guest_Template struct {
 	//Required
 	Hostname          string     `json:"hostname"`
 	Domain            string     `json:"domain"`
 	StartCpus         int        `json:"startCpus"`
 	MaxMemory         int        `json:"maxMemory"`
+
 	Datacenter        Datacenter `json:"datacenter"`
 	HourlyBillingFlag bool       `json:"hourlyBillingFlag"`
 	LocalDiskFlag     bool       `json:"localDiskFlag"`
-
-	//Conditionally required
-	OperatingSystemReferenceCode string                    `json:"operatingSystemReferenceCode,omitempty"`
-	BlockDeviceTemplateGroup     *BlockDeviceTemplateGroup `json:"blockDeviceTemplateGroup,omitempty"`
-
-	//Optional
 	DedicatedAccountHostOnlyFlag   bool                            `json:"dedicatedAccountHostOnlyFlag,omitempty"`
```

Comment:
Why did you remove the Optional comment. These are the type of comments that are good in code. Simple: `//Required`, `//Optional`, etc and maybe a simple link to documentation.